### ISSUE TITLE
Reopen dialog on openedChanged event makes the dialog broken.

### ISSUE THREAD

**Issue by alexberazouski** [vaadin/vaadin-dialog-flow#101]
Consider the following example. No matter dialog is self-attached or is inside of template bound by `@Id`:
```java
dialog.addOpenedChangeListener(e -> {
        if (dialog.isOpened()) return;
                                
        if (someCondition) dialog.setOpened(true);
});
```
Under some condition the dialog should be reopened after it was closed.

In browser I see weird placeholders that appear on each reopen.
```html
<!--vaadin-overlay-placeholder-->
<!--vaadin-overlay-placeholder-->
<!--vaadin-overlay-placeholder-->
<!--vaadin-overlay-placeholder-->
<vaadin-dialog-overlay id="overlay" with-backdrop="" focus-trap="" 
  role="dialog" opened=""><vaadin-dialog-overlay>
```

More over, the `vaadin-dialog-overlay` stays in Dom even if the dialog itself is finally closed and detached.

Looks like some timing issue, cause if I have a stop in a debugger before `setOpened(true)`, issue is not reproduced.

It brings a bigger problem if instead of `if (someCondition) dialog.setOpened(true);` there is  another dialog opened (e.g. confirmation dialog use case). In that case I can observe an overlay with `backdrop` on top of everything, which makes it impossible to interact with anything in the dialog overlay.

Basically, if the dialog once was reopened in the way described above it becomes broken until page reload.

Yes, I am aware of `DialogCloseActionEvent`, and it helped in my case.

**Comment by Artur-** [comment]
https://github.com/vaadin/vaadin-dialog/issues/111 ?
